packages/palette/src/elements/ModalDialog/ModalDialogDialog.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react"
+import { Clickable } from "../Clickable"
+import { DROP_SHADOW } from "../../helpers/shadow"
+import { Box, BoxProps } from "../Box"
+import { Flex } from "../Flex"
+import { Text } from "../Text"
+import { CloseIcon } from "../../svgs/CloseIcon"
+import { ArtsyLogoBlackIcon } from "../../svgs/ArtsyLogoBlackIcon"
+import { useSentinelVisibility } from "../../utils/useSentinelVisibility"
+import { Spacer } from "../Spacer"
+
+export interface ModalDialogDialogProps extends BoxProps {
+  children: React.ReactNode
+  footer?: React.ReactNode
+  hasLogo?: boolean
+  onClose: () => void
+  title?: string
+}
+
+export const ModalDialogDialog: React.FC<ModalDialogDialogProps> = ({
+  children,
+  footer,
+  hasLogo,
+  onClose,
+  title,
+  ...rest
+}) => {
+  const {
+    sentinel: topSentinel,
+    isSentinelVisible: isAtTop,
+  } = useSentinelVisibility()
+
+  const {
+    sentinel: bottomSentinel,
+    isSentinelVisible: isAtBottom,
+  } = useSentinelVisibility()
```

Comment:
There are two shadows that we independently turn on and off.

> Shadows, similar to material, only appears as content goes behind.

Reply:
Whats a good dead-clear synonym for "sentinel"

Reply:
Sentinel is pretty clear and a commonly used word for this sort of thing: https://developers.google.com/web/updates/2016/04/intersectionobserver — this is also the nomenclature Google uses.

Reply:
Cool, lets stick with google defs 👍